lib/elixir/pages/anti-patterns/design-anti-patterns.md

```diff
@@ -69,7 +69,62 @@ The following arguments were given to MyLibrary.foo/1:
 
 ## Alternative return types
 
-TODO
+#### Problem
+
+This anti-pattern refers to functions that receive options (for example, *keyword list*) parameters that drastically change their return type. Because options are optional and sometimes set dynamically, if they change the return type it may be hard to understand what the function actually returns.
```

Comment:
Let's not mention "options are optional" (that's not true, they only are *sometimes*):

```suggestion
This anti-pattern refers to functions that take options (for example, in the form of a *keyword list*) through parameters that drastically change their return type. This requires developers to know the value of those options at runtime for every call in order to determine what the return type of the function is, making the code harder to understand.
```